pkg/catalog/catalog.go

```diff
@@ -1175,6 +1180,77 @@ func (c *Catalog) GetRange(ctx context.Context, repositoryID, rangeID string) (g
 	return c.Store.GetRange(ctx, graveler.RepositoryID(repositoryID), graveler.RangeID(rangeID))
 }
 
+func (c *Catalog) GetRetentionRules(ctx context.Context, repositoryID string) (*graveler.RetentionRules, error) {
+	return c.Store.GetRetentionRules(ctx, graveler.RepositoryID(repositoryID))
+}
+
+func (c *Catalog) SetRetentionRules(ctx context.Context, repositoryID string, rules *graveler.RetentionRules) error {
+	return c.Store.SetRetentionRules(ctx, graveler.RepositoryID(repositoryID), rules)
+}
+
+func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repository string, previousResultPath string) (string, error) {
+	repositoryID := graveler.RepositoryID(repository)
+	if err := Validate([]ValidateArg{
+		{"repositoryID", repositoryID, ValidateRepositoryID},
+	}); err != nil {
+		return "", err
+	}
+	repo, err := c.Store.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return "", err
+	}
+	previouslyExpiredCommits := make([]graveler.CommitID, 0)
+	if previousResultPath != "" {
+		previousRunReader, err := c.BlockAdapter.Get(ctx, block.ObjectPointer{
+			StorageNamespace: string(repo.StorageNamespace),
+			Identifier:       previousResultPath,
+			IdentifierType:   block.IdentifierTypeRelative,
+		}, -1)
+		if err != nil {
+			return "", err
+		}
+		csvReader := csv.NewReader(previousRunReader)
+		previousCommits, err := csvReader.ReadAll()
+		if err != nil {
+			return "", err
+		}
+		for _, commitRow := range previousCommits {
+			previouslyExpiredCommits = append(previouslyExpiredCommits, graveler.CommitID(commitRow[1]))
+		}
+	}
+	activeCommits, expiredCommits, err := c.Store.GetExpiredCommits(ctx, repositoryID, previouslyExpiredCommits)
+	if err != nil {
+		return "", fmt.Errorf("preparing expired commits: %w", err)
+	}
+	b := &strings.Builder{}
+	csvWriter := csv.NewWriter(b)
+	for _, commitID := range expiredCommits {
+		err = csvWriter.Write([]string{string(commitID), strconv.FormatBool(true)})
+		if err != nil {
+			return "", err
+		}
+	}
+	for _, commitID := range activeCommits {
+		err = csvWriter.Write([]string{string(commitID), strconv.FormatBool(false)})
+		if err != nil {
+			return "", err
+		}
+	}
+	csvWriter.Flush()
+	commitsStr := b.String()
+	runID := uuid.New().String()
+	path := fmt.Sprintf("_lakefs/retention/commits/run_id=%s/commits.csv", runID)
```

Comment:
`_lakefs` prefix should be passed to you from the config.

Reply:
Maybe add the `/retention/commits/` part too.

Reply:
Done the first one